src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,63 @@
+import { deleteTodo } from '../../api/todos';
+import { Errors } from '../../enums/Errors';
+import { useTodosContext } from '../../hooks/useTodosContext';
+import { TodoFilter } from '../TodoFilter';
+
+export const Footer: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    setLoadingTodoIds,
+    showError,
+    setIsFocusedInput,
+    notCompletedCount,
+    completedCount,
+  } = useTodosContext();
+
+  const handleDeleteTodo = async (todoId: number): Promise<void> => {
+    try {
+      await deleteTodo(todoId);
+      setTodos(prevTodos => prevTodos.filter(t => t.id !== todoId));
+      setLoadingTodoIds([]);
+    } catch (err) {
+      showError(Errors.DeleteTodo);
+      setLoadingTodoIds([]);
+    }
+  };
+
+  const deleteCompletedTodos = () => {
+    const completedTodoIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    setLoadingTodoIds(completedTodoIds);
```

Comment:
And when you add new loading todo ids the same
`setLoadingTodoIds(prevLoadingTodosIds => [...prevLoadingTodosIds, ...completedTodoIds])`